File: web/src/api/expertAcademicPosition.js
import service from '@/utils/request'

// @Tags ExpertAcademicPosition
// @Summary 创建专家学术兼职
// @Router /expertAcademicPosition/createExpertAcademicPosition [post]
export const createExpertAcademicPosition = (data) => {
  return service({
    url: '/expertAcademicPosition/createExpertAcademicPosition',
    method: 'post',
    data
  })
}

// @Tags ExpertAcademicPosition
// @Summary 删除专家学术兼职
// @Router /expertAcademicPosition/deleteExpertAcademicPosition [delete]
export const deleteExpertAcademicPosition = (params) => {
  return service({
    url: '/expertAcademicPosition/deleteExpertAcademicPosition',
    method: 'delete',
    params
  })
}

// @Tags ExpertAcademicPosition
// @Summary 批量删除专家学术兼职
// @Router /expertAcademicPosition/deleteExpertAcademicPositionByIds [delete]
export const deleteExpertAcademicPositionByIds = (params) => {
  return service({
    url: '/expertAcademicPosition/deleteExpertAcademicPositionByIds',
    method: 'delete',
    params
  })
}

// @Tags ExpertAcademicPosition
// @Summary 更新专家学术兼职
// @Router /expertAcademicPosition/updateExpertAcademicPosition [put]
export const updateExpertAcademicPosition = (data) => {
  return service({
    url: '/expertAcademicPosition/updateExpertAcademicPosition',
    method: 'put',
    data
  })
}

// @Tags ExpertAcademicPosition
// @Summary 用id查询专家学术兼职
// @Router /expertAcademicPosition/findExpertAcademicPosition [get]
export const findExpertAcademicPosition = (params) => {
  return service({
    url: '/expertAcademicPosition/findExpertAcademicPosition',
    method: 'get',
    params
  })
}

// @Tags ExpertAcademicPosition
// @Summary 分页获取专家学术兼职列表
// @Router /expertAcademicPosition/getExpertAcademicPositionList [get]
export const getExpertAcademicPositionList = (params) => {
  return service({
    url: '/expertAcademicPosition/getExpertAcademicPositionList',
    method: 'get',
    params
  })
}

File: server/router/ExpertDatabase/expert_academic_position.go
package ExpertDatabase

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ExpertAcademicPositionRouter struct{}

// InitExpertAcademicPositionRouter 初始化 专家学术兼职 路由信息
func (s *ExpertAcademicPositionRouter) InitExpertAcademicPositionRouter(Router *gin.RouterGroup) {
	expertAcademicPositionRouter := Router.Group("expertAcademicPosition").Use(middleware.OperationRecord())
	expertAcademicPositionRouterWithoutRecord := Router.Group("expertAcademicPosition")
	expertAcademicPositionApi := v1.ApiGroupApp.ExpertDatabaseApiGroup.ExpertAcademicPositionApi
	{
		expertAcademicPositionRouter.POST("createExpertAcademicPosition", expertAcademicPositionApi.CreateExpertAcademicPosition)             // 新建专家学术兼职
		expertAcademicPositionRouter.DELETE("deleteExpertAcademicPosition", expertAcademicPositionApi.DeleteExpertAcademicPosition)           // 删除专家学术兼职
		expertAcademicPositionRouter.DELETE("deleteExpertAcademicPositionByIds", expertAcademicPositionApi.DeleteExpertAcademicPositionByIds) // 批量删除专家学术兼职
		expertAcademicPositionRouter.PUT("updateExpertAcademicPosition", expertAcademicPositionApi.UpdateExpertAcademicPosition)              // 更新专家学术兼职
	}
	{
		expertAcademicPositionRouterWithoutRecord.GET("findExpertAcademicPosition", expertAcademicPositionApi.FindExpertAcademicPosition)       // 根据ID获取专家学术兼职
		expertAcademicPositionRouterWithoutRecord.GET("getExpertAcademicPositionList", expertAcademicPositionApi.GetExpertAcademicPositionList) // 分页获取专家学术兼职列表
	}
}
